vllm/entrypoints/openai/protocol.py

```diff
@@ -60,10 +68,31 @@ class ResponseFormat(BaseModel):
     type: str = Literal["text", "json_object"]
 
 
+class TextContent(BaseModel):
+    type: str = Literal["text"]
+    text: str
+
+
+SingleTextContentList = Annotated[List[TextContent], Len(1, 1)]
+
+
+def try_flatten_text_content(value: any) -> str:
+    # The OpenAI API allows complex contents in chat messages. While
+    # we do not fully support them at the moment, we can allow simple
+    # cases where just a single text content is passed in a message.
+    with contextlib.suppress(ValidationError):
```

Comment:
I am a bit concerned here, the validator is triggered for each request. The validator is strict and contains multiple steps. I'm not sure how much overhead will it introduce. Any comments or thoughts? @simon-mo @fgreinacher

Reply:
I don't expect this to be a problem. Validation happens anyway for every request and this just does a bit more. But we could optimize a bit by skipping this validation if `value` is a plain string.

Reply:
I have pushed that optimization. With that performance will be unchanged for simple text-only requests.

Reply:
Yeah looks good to me now. Please just fix the conflict and I will merge it, thanks!

Reply:
Thanks @LiuXiaoxuanPKU, I have rebased the PR so this should be good to go 🙇